docs/src/pages/versions/versions.md

```diff
@@ -17,19 +17,38 @@ You can use it to see what changes are coming and provide better feedback to MUI
 
 MUI follows [Semantic Versioning 2.0.0](https://semver.org/).
 MUI version numbers have three parts: `major.minor.patch`.
 The version number is incremented based on the level of change included in the release.
 
-- **Major releases** contain significant new features, some but minimal developer assistance is expected during the update. When updating to a new major release, you may need to run update scripts, refactor code, run additional tests, and learn new APIs.
+- **Major releases** contain significant new features, some but minimal developer assistance is expected during the update.
+  These releases include [breaking changes](#what-doesnt-count-as-a-breaking-change).
+  When updating to a new major release, you may need to run update scripts, refactor code, run additional tests, and learn new APIs.
 - **Minor releases** contain important new features.
-  Minor releases are fully backward-compatible; no developer assistance is expected during update, but you can optionally modify your apps and libraries to begin using new APIs, features, and capabilities that were added in the release.
+  Minor releases are fully backward-compatible; no developer assistance is expected during the update, but you can optionally modify your apps and libraries to begin using new APIs, features, and capabilities that were added in the release.
 - **Patch releases** are low risk, contain bug fixes and small new features.
-  No developer assistance is expected during update.
+  No developer assistance is expected during the update.
+
+## What doesn't count as a breaking change?
+
+We call "breaking changes" those that require updating your codebase when upgrading to a new version, with the exception of:
+
+- **APIs starting with "unstable\_"**. These are provided as experimental features whose APIs we are not yet confident in.
+  By releasing these with an `unstable_` prefix, we can iterate faster and get to a stable API sooner, or simply learn that we don't need the API/feature in the first place.
+- **APIs documented as experimental**. Same as the above.
+- **Undocumented APIs and internal data structures**. If you access internal properties, there is no warranty. You are on your own.
+- **Development warnings**. Since these don't affect production behavior, we may add new warnings or modify existing warnings in between major versions.
+  In fact, this is what allows us to reliably warn about upcoming breaking changes.
+- **Pre-releases versions**. We provide pre-release versions as a way to test new features early, but we need the flexibility to make changes based on what we learn in the pre-release period.
+  If you use these versions, note that APIs may change before the stable release.
+- **Small CSS changes**. Visual design changes that have a very low probability of negatively impacting your UI are not considered breaking.
+
```

Comment:
What about supported node version?
We have this section on Supported platforms: https://mui.com/material-ui/getting-started/supported-platforms/#server

But node v12 is not supported anymore (v14 is in maintenance LTS) and I've opened a PR in MUI X: https://github.com/mui/mui-x/pull/4999

Side note:
> MUI supports [Node.js](https://github.com/nodejs/node) starting with version 12.17 (or 12.0 with --experimental-modules enabled) for server-side rendering

I don't understand how `--experimental-modules` flag is relevant here, because we provide `node` target folder that uses CommonJS (see https://unpkg.com/@mui/material@5.8.0/node/index.js)

Reply:
> I don't understand how --experimental-modules flag is relevant here, because we provide node target folder that uses CommonJS

I was mostly to allow us to switch to ES modules in a minor release. But since then it's become clear that the migration strategy offered by Node is not suitable and every package should switch to ES modules in a major release.